src/bci_build/package/apache_tomcat.py

```diff
@@ -12,6 +13,36 @@
 _TOMCAT_VERSIONS: list[int] = [9, 10]
 assert _TOMCAT_VERSIONS == sorted(_TOMCAT_VERSIONS)
 
+
+def _get_sac_supported_until(
+    os_version: OsVersion, tomcat_major: int, jre_major: int
+) -> datetime.date:
+    """Return the prodicted minimum end of support date for this os/tomcat/jre combination."""
+    if os_version.is_tumbleweed:
+        return None
+
+    # Taken from https://www.suse.com/releasenotes/x86_64/SUSE-SLES/15-SP6/index.html#java-version
+    jre_end_support_dates: dict[int, datetime.date] = {
+        21: datetime.date(2031, 6, 30),
+        17: datetime.date(2027, 12, 31),
+        11: datetime.date(2026, 12, 31),
+    }
+
+    # We do not have a documented policy, suggestion is to do 3 years from initial
+    # publishing
+    tomcat_end_support_dates: dict[int, datetime.date] = {
+        9: datetime.date(2024 + 3, 2, 1),
+        10: datetime.date(2024 + 3, 7, 1),
+    }
+    end_date: datetime.date = min(
+        jre_end_support_dates.get(jre_major, datetime.date.max),
+        tomcat_end_support_dates.get(tomcat_major, datetime.date.max),
+    )
+    if end_date < datetime.date.max:
+        return end_date
```

Comment:
datetime.date.max is the year 9999, why check against that?

Reply:
we are checking here that either tomcat or openjdk had a time set (because of the default attribute in get() call above. if we did then we return the minimum (aka earliest date) of tomcat+openjdk combination.

Reply:
okay, now I understand this. Can you please add a comment explaining this briefly? Or maybe refactor the code a bit so that this is clearer, e.g.:
```Python
# we cannot infer a eol date either from tomcat, nor from openjdk
if jre_major not in jre_end_support_dates and tomcat_major not in tomcat_end_support_dates:
    return None

# pick the lower eol of tomcat & openjdk
end_date = min(
        jre_end_support_dates.get(jre_major, datetime.date.max),
        tomcat_end_support_dates.get(tomcat_major, datetime.date.max),
    )
assert end_date < datetime.date.max
return end_date
```

Reply:
I did have it briefly documented in the function/method docstring, but it is expanded now to be more verbose.